File: presentation/src/main/java/com/lanit_tercom/dogfriendly_studproject/MainActivity.java
package com.lanit_tercom.dogfriendly_studproject;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;

import com.lanit_tercom.data.entity.UserEntity;
import com.lanit_tercom.data.firebase.FirebaseUserEntityStore;
import com.lanit_tercom.data.firebase.cache.UserCache;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //Test getUserById() and getAllUsers() methods from FirebaseUserEntityStore
        FirebaseUserEntityStore firebaseUserEntityStore = new FirebaseUserEntityStore(new UserCache() {
            @Override
            public void saveUser(String userId, UserEntity userEntity) {

            }

            @Override
            public void getAllUsers(UserListCallback userListCallback) {

            }

            @Override
            public void getUserById(String id, UserByIdCallback userByIdCallback) {

            }
        });
    }
}

File: presentation/src/main/java/com/lanit_tercom/dogfriendly_studproject/ui/activity/MainActivity.kt
package com.lanit_tercom.dogfriendly_studproject.ui.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.lanit_tercom.dogfriendly_studproject.R
import com.lanit_tercom.dogfriendly_studproject.ui.fragment.OnBackButtonListener
import com.lanit_tercom.dogfriendly_studproject.ui.fragment.UserSignInFragment

/**
 * Главная активность.
 * Запускает фрагмент с картой.
 * @author nikolaygorokhov1@gmail.com
 * @author prostak.sasha111@mail.ru
 *
 */
class MainActivity : BaseActivity(){

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val signInFragment = UserSignInFragment()

        if (savedInstanceState == null)
            addFragment(R.id.ft_container, signInFragment)
    }

    override fun onBackPressed() {
        val backStackCount = supportFragmentManager.backStackEntryCount

        if (backStackCount > 0) {
            val currentFragment: Fragment? = supportFragmentManager.findFragmentById(R.id.ft_container)
            if (currentFragment is OnBackButtonListener){
                if (currentFragment.onBackPressed()) return
            }
        }

        super.onBackPressed()
    }
}